From 374b73d4d1f2d7c7ab318725482f4712726307ec Mon Sep 17 00:00:00 2001
From: Kashish Awasthi <quic_kawasthi@quicinc.com>
Date: Wed, 6 Nov 2024 17:15:39 +0530
Subject: [PATCH] Add a new QCA vendor attribute for TWT session suspendability

Add the following new QCA vendor attribute to configure whether
the TWT session can be suspended:
QCA_WLAN_VENDOR_ATTR_TWT_SETUP_SUSPENDABLE

Signed-off-by: Kashish Awasthi <quic_kawasthi@quicinc.com>
---
 src/common/qca-vendor.h | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/src/common/qca-vendor.h b/src/common/qca-vendor.h
index 170fb62c7..bf98890c1 100644
--- a/src/common/qca-vendor.h
+++ b/src/common/qca-vendor.h
@@ -10832,6 +10832,18 @@ enum qca_wlan_twt_setup_state {
 	QCA_WLAN_TWT_SETUP_STATE_SUSPEND = 2,
 };
 
+/**
+ * enum qca_wlan_twt_session_suspendable: The values used with
+ * %QCA_WLAN_VENDOR_ATTR_TWT_SETUP_SUSPENDABLE.
+ *
+ * @QCA_WLAN_TWT_SESSION_NOT_SUSPENDABLE: TWT session cannot be suspended.
+ * @QCA_WLAN_TWT_SESSION_SUSPENDABLE: TWT session can be suspended.
+ */
+enum qca_wlan_twt_session_suspendable {
+	QCA_WLAN_TWT_SESSION_NOT_SUSPENDABLE = 0,
+	QCA_WLAN_TWT_SESSION_SUSPENDABLE = 1,
+};
+
 /**
  * enum qca_wlan_vendor_attr_twt_setup: Represents attributes for
  * TWT (Target Wake Time) setup request. These attributes are sent as part of
@@ -11061,7 +11073,15 @@ enum qca_wlan_twt_setup_state {
  *
  * @QCA_WLAN_VENDOR_ATTR_TWT_SETUP_PAD: Attribute used for padding for 64-bit
  * alignment.
+ *
+ * @QCA_WLAN_VENDOR_ATTR_TWT_SETUP_SUSPENDABLE: Optional (u8)
+ * This attribute indicates whether the TWT session being negotiated can be
+ * suspended.
+ * Refers the enum qca_wlan_twt_session_suspendable.
+ * This parameter is used for
+ * 1. TWT SET Response
  */
+
 enum qca_wlan_vendor_attr_twt_setup {
 	QCA_WLAN_VENDOR_ATTR_TWT_SETUP_INVALID = 0,
 	QCA_WLAN_VENDOR_ATTR_TWT_SETUP_BCAST = 1,
@@ -11098,6 +11118,7 @@ enum qca_wlan_vendor_attr_twt_setup {
 	QCA_WLAN_VENDOR_ATTR_TWT_SETUP_ANNOUNCE_TIMEOUT = 26,
 
 	QCA_WLAN_VENDOR_ATTR_TWT_SETUP_PAD = 27,
+	QCA_WLAN_VENDOR_ATTR_TWT_SETUP_SUSPENDABLE = 28,
 
 	/* keep last */
 	QCA_WLAN_VENDOR_ATTR_TWT_SETUP_AFTER_LAST,
-- 
2.43.0

